components/mediation-ntask/src/main/java/org/wso2/carbon/mediation/ntask/NTaskTaskManager.java

```diff
@@ -533,88 +548,88 @@ private String managerId() {
         return "[NTaskTaskManager::" + getCurrentTenantId() + " ::" + this.hashCode() + "]";
     }
 
-	@Override
-	public void addObserver(TaskManagerObserver o) {
-		if (observers.contains(o)) {
-			return;
-		}
-		observers.add(o);
-	}
+    @Override
+    public void addObserver(TaskManagerObserver o) {
+        if (observers.contains(o)) {
```

Comment:
This was fixed in pull request https://github.com/wso2/carbon-mediation/pull/258